static/js/subs.js

```diff
@@ -366,6 +374,10 @@ exports.mark_sub_unsubscribed = function (sub) {
             settings.collapse('hide');
         }
 
+        // Update subscribers count
+        var count = stream_data.get_subscribers_count(sub.name) - 1;
```

Comment:
I'm pretty sure this is intentional behavior -- Zulip keeps track of information about streams the user used to be subscribed to but isn't anymore, and that's what leads to these subscription elements with `sub.subscribed = false`.  You just shouldn't include those subscription objects as indicating the user is subscribed to the stream, hence my suggestion that you use slightly more complex code to only count those subscribers who have `sub.subscribed === true`